### PR TITLE
chore: minor ui fixes in tooltip

### PR DESCRIPTION
**Pull request by ahrefabhi**
## Pull Request

---

### 📄 Summary

This PR refactors how the tooltip footer is composed across dashboard chart panels (TimeSeries, Bar, Histogram). Previously, the tooltip footer was rendered unconditionally inside the generic `uPlotV2` `Tooltip` component, controlled by a boolean `canPinTooltip` flag. This made it impossible for individual panels to customise footer behaviour (e.g. hiding the drilldown hint for Histogram panels).

The changes introduce a `renderTooltipFooter` render prop on `BaseTooltipProps` and `BaseChartProps`, so each panel supplies its own footer renderer. `TooltipFooter` is moved out of the `uPlotV2` library tree into the `DashboardContainer/visualization/panels/components/` layer, where it belongs. As a side-effect, all tooltip-related analytics events now emit the **panel ID** (`config.getId()`) instead of the current URL path, giving more precise attribution per panel rather than per page.

Minor CSS fixes are included to correct spacing and divider rendering in the tooltip.

---

image

### No drilldown in Histogram Panel
image

---

---

### ✅ Change Type

- [ ] ✨ Feature
- [ ] 🐛 Bug fix
- [x] ♻️ Refactor
- [ ] 🛠️ Infra / Tooling
- [ ] 🧪 Test-only

---

---

### 🧪 Testing Strategy

- Tests added/updated: `Tooltip.test.tsx` updated to include required `id` prop
- Manual verification: Tooltip pin/unpin, drilldown hint visibility, footer rendering across TimeSeries, Bar, and Histogram panels
- Edge cases covered: Histogram panel correctly suppresses drilldown hint via `canDrilldown={false}`

---

### ⚠️ Risk & Impact Assessment

- Blast radius: Dashboard tooltip rendering in all chart panel types (TimeSeries, Bar, Histogram)
- Potential regressions: Any custom `customTooltip` usages that relied on `canPinTooltip` — that prop is now superseded by `renderTooltipFooter`; the old `pinnedTooltipElement` prop on `TimeSeries` is removed
- Rollback plan: Revert the PR; no backend changes

---

### 📝 Changelog

| Field | Value |
|------|-------|
| Deployment Type | Cloud / OSS / Enterprise |
| Change Type | Maintenance |
| Description | N/A — internal refactor of tooltip footer composition; no user-visible behaviour change beyond minor spacing fixes |

---

### 📋 Checklist
- [x] Tests added or explicitly not required
- [x] Manually tested
- [x] Breaking changes documented
- [x] Backward compatibility considered

---

## 👀 Notes for Reviewers

- `TooltipFooter` is intentionally moved to the panel layer — it references `widget.id` and panel-specific event logic, so it was never really a generic uPlot primitive.
- Analytics events now emit `id: panelId` instead of `path: window.location.pathname`; this is a deliberate improvement (page path was too coarse for multi-panel dashboards).
- `pinnedTooltipElement` prop on `TimeSeries` has been **removed** — if anything outside this PR was passing it, it will now be silently ignored at the type level. Double-check usages if unsure.

### CODE REVIEW

File: frontend/src/container/DashboardContainer/visualization/panels/components/TooltipFooter.tsx [hunk 1]
```diff
@@ -43,12 +46,14 @@ export default function TooltipFooter({
...
-						</div>
+						{canDrilldown && (
+							<div className={Styles.hint} data-active="false">
+								<Kbd>
+									<MousePointerClick size={12} />
+								</Kbd>
+								<span>Click to drilldown</span>
+							</div>
+						)}
 						<div className={Styles.hint} data-active="false">
```
Comment (AshwinBhatkal) [inline frontend/src/container/DashboardContainer/visualization/panels/components/TooltipFooter.tsx:57]: Should we have canPin here?

Comment (AshwinBhatkal) [reply frontend/src/container/DashboardContainer/visualization/panels/components/TooltipFooter.tsx:57]: For cases outside dashboard?

Comment (AshwinBhatkal) [reply frontend/src/container/DashboardContainer/visualization/panels/components/TooltipFooter.tsx:57]: The tooltip footer is no more just for pinning

File: frontend/src/lib/uPlotV2/components/Tooltip/__tests__/Tooltip.test.tsx [hunk 2]
```diff
@@ -192,62 +193,87 @@ describe('Tooltip', () => {
...
+		);
+
+		renderTooltip({ renderTooltipFooter, isPinned: true, dismiss });
 
-		const footer = screen.getByRole('status');
-		expect(footer).toBeInTheDocument();
+		const user = userEvent.setup();
+		await user.click(screen.getByTestId('dismiss-btn'));
+
+		expect(dismiss).toHaveBeenCalledTimes(1);
```
Comment (AshwinBhatkal) [inline frontend/src/lib/uPlotV2/components/Tooltip/__tests__/Tooltip.test.tsx:276]: Can we add test for TooltipFooter? The current test tests rendertooltip

File: frontend/src/container/DashboardContainer/visualization/panels/TimeSeriesPanel/TimeSeriesPanel.tsx [hunk 3]
```diff
@@ -104,8 +108,16 @@ function TimeSeriesPanel(props: PanelWrapperProps): JSX.Element {
...
 
+	const renderTooltipFooter = useCallback(
+		({ isPinned, dismiss }: IRenderTooltipFooterArgs) => {
+			return (
+				<TooltipFooter id={widget.id} isPinned={isPinned} dismiss={dismiss} />
+			);
+		},
+		[],
+	);
 	const groupBy = useMemo(() => {
```
Comment (AshwinBhatkal) [inline frontend/src/container/DashboardContainer/visualization/panels/TimeSeriesPanel/TimeSeriesPanel.tsx]: It may have been here. But given the current context that I have, this should not be here. It should passed as a prop to the panelWrapper itself. 
How does groupBy in a promQL or clickhouseSQL query translate to the chart?

Comment (ahrefabhi) [reply frontend/src/container/DashboardContainer/visualization/panels/TimeSeriesPanel/TimeSeriesPanel.tsx]: We can't identify groupby in promql and clickhouse query

File: frontend/src/container/DashboardContainer/visualization/charts/ChartWrapper/ChartWrapper.tsx [hunk 4]
```diff
@@ -33,7 +33,7 @@ export default function ChartWrapper({
 	children,
 	layoutChildren,
 	yAxisUnit,
-	groupBy,
+	groupByPerQuery,
```
Comment (AshwinBhatkal) [inline frontend/src/container/DashboardContainer/visualization/charts/ChartWrapper/ChartWrapper.tsx:36]: perPanel noe?

File: frontend/src/container/PanelWrapper/PanelWrapper.tsx [hunk 5]
```diff
@@ -30,6 +31,17 @@ function PanelWrapper({
 		selectedGraph || widget.panelTypes
 	] as FC<PanelWrapperProps>;
 
+	const groupByPerQuery = useMemo<Record<string, BaseAutocompleteData[]>>(() => {
+		const { queryData } = widget.query.builder;
+		return queryData.reduce<Record<string, BaseAutocompleteData[]>>(
```
Comment (AshwinBhatkal) [inline frontend/src/container/PanelWrapper/PanelWrapper.tsx:39]: will queryData always be an array?

File: frontend/src/container/PanelWrapper/PanelWrapper.tsx [hunk 6]
```diff
@@ -30,6 +31,17 @@ function PanelWrapper({
...
+			(acc, query) => {
+				acc[query.queryName] = query.groupBy ?? [];
```
Comment (AshwinBhatkal) [inline frontend/src/container/PanelWrapper/PanelWrapper.tsx:41]: will queryName always be there?

File: frontend/src/lib/uPlotV2/plugins/TooltipPlugin/types.ts [hunk 7]
```diff
@@ -40,7 +40,7 @@ export interface TooltipLayoutInfo {
 
 export interface TooltipSyncMetadata {
 	yAxisUnit?: string;
-	groupBy?: BaseAutocompleteData[];
+	groupByPerQuery?: Record<string, BaseAutocompleteData[]>;
```
Comment (AshwinBhatkal) [inline frontend/src/lib/uPlotV2/plugins/TooltipPlugin/types.ts:43]: what about groupByForPanelQueries? It could be an array. I could not see a use for the keys

**Review by AshwinBhatkal — APPROVED** [review]
